launch_xml/launch_xml/parser.py

```diff
@@ -33,3 +35,8 @@ def load(
     ) -> (Entity, 'Parser'):
         """Return entity loaded from XML file."""
         return (Entity(ET.parse(file).getroot()), cls())
+
+    @classmethod
+    def get_file_extensions(cls) -> Set[Text]:
+        """Return the set of file extensions known to this parser."""
+        return {'launch.xml'}
```

Comment:
maybe we should add also `xml` here.
I know there are packages that are naming launch files like `_launch.xml`.

Reply:
we could also test for `'_' + ext`, but I'll just add `xml`: 74934f64093f6e7d9d75cb38c31e5c6e61ea6c5c

Reply:
We should probably do the same for `yaml`, no?

Reply:
> we could also test for '_' + ext, but I'll just add xml: 74934f6

:+1:
I think we should do the same for yaml.

Reply:
> :+1:

Is this a :+1: for checking for `filename.endswith('_' + ext)` as well as `filename.endswith('.' + ext)`? :sweat_smile:

> I think we should do the same for yaml.

Should I also add `yml`?

Reply:
> Is this a +1 for checking for filename.endswith('_' + ext) as well as filename.endswith('.' + ext)? sweat_smile

Sorry, I meant the current commit was ok (i.e. adding xml and not only launch.xml to the accepted extensions).

> I'll add it. Should I also add yml?

yes, for all extensions we should support .launch.ext and .ext IMO.

Reply:
done in aa3b33144dc42763b178876ed845f387b163f4e9

Reply:
> yes, for all extensions we should support .launch.ext and .ext IMO.

Makes sense. Strictly implementation-wise, it is kind of redundant, but explicitly writing the extensions out like that makes sense.

Reply:
> I know there are packages that are naming launch files like `_launch.xml`.

I even recently saw one that uses `-launch.py` (https://github.com/KumarRobotics/imu_vn_100/blob/dashing/launch/imu_vn_100-launch.py), but I think that's from before there was any kind of discussion about picking a standard extension.